packages/select/index.tsx

```diff
@@ -19,48 +19,83 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-
-import React from 'react';
-import PropTypes from 'prop-types';
-import classnames from 'classnames';
+import * as React from 'react';
+import * as classnames from 'classnames';
+// no mdc .d.ts file
+// @ts-ignore
 import {MDCSelectFoundation} from '@material/select/dist/mdc.select';
-
 import FloatingLabel from '@material/react-floating-label';
 import LineRipple from '@material/react-line-ripple';
 import NotchedOutline from '@material/react-notched-outline';
 import NativeControl from './NativeControl';
 
-export default class Select extends React.Component {
-  foundation_ = null;
+export interface SelectProps extends React.HTMLProps<HTMLSelectElement> {
+  box: boolean;
+  className: string;
+  disabled: boolean;
+  floatingLabelClassName?: string;
+  isRtl: boolean;
+  label: string;
+  lineRippleClassName: string;
+  nativeControlClassName: string;
+  notchedOutlineClassName: string;
+  outlined: boolean;
+  options: (string | React.HTMLProps<HTMLOptionElement>)[];
```

Comment:
when would options be a string vs an element? Can it be restricted to make its handling easier?

Reply:
If people just want to do 
```
options = ['poodle', 'golden_retreiver', 'alaskan']

or

options = [{
  label: 'Poodle', value: 'poodle'
}, {
 label: 'Golden Retriever', value: 'golden_retriever'
}, ...]
```

Any suggestions? If we changed it, it would be a breaking change